### PR TITLE
chore: partition by null should set key format to NONE

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

fixes: https://github.com/confluentinc/ksql/issues/6391

### Testing done 

Usual

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by vcrfxia — APPROVED** [review]
LGTM. Thanks!

File: ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java [hunk 1]
```diff
@@ -189,6 +191,30 @@ private void analyzeNonStdOutSink(final Sink sink) {
...
+      if (partitioningByNull) {
+        final boolean nonNoneExplicitFormat = explicitFormat
+            .map(fmt -> !fmt.equalsIgnoreCase(NoneFormat.NAME))
+            .orElse(false);
+
+        if (nonNoneExplicitFormat) {
+          throw new KsqlException("Key format specified for stream without key columns.");
+        }
+
+        FormatInfo.of(NoneFormat.NAME);
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java:212]: did you mean to return here?

File: ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java [hunk 2]
```diff
@@ -189,6 +191,30 @@ private void analyzeNonStdOutSink(final Sink sink) {
...
+          .map(pb -> pb.getExpression() instanceof NullLiteral)
+          .orElse(false);
+
+      if (partitioningByNull) {
+        final boolean nonNoneExplicitFormat = explicitFormat
+            .map(fmt -> !fmt.equalsIgnoreCase(NoneFormat.NAME))
+            .orElse(false);
+
+        if (nonNoneExplicitFormat) {
+          throw new KsqlException("Key format specified for stream without key columns.");
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java:209]: why is this an error? can't I have `null` keys with non-`NONE` formats?

Comment (big-andy-coates) [reply ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java:209]: If the key format supports schema inference then it's going to blow up later when trying to build the schema for the key.

File: ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java (hunk 1 above)
Comment (big-andy-coates) [reply ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java:212]: Yup!

**Pull request merged**